extensions-core/lookups-cached-single/src/main/java/io/druid/server/lookup/PollingLookup.java

```diff
@@ -107,8 +108,14 @@ public void close()
   }
 
   @Override
-  public String apply(@NotNull String key)
+  @Nullable
+  public String apply(@Nullable String key)
   {
+    String keyEquivalent = NullHandling.nullToEmptyIfNeeded(key);
+    if (keyEquivalent == null) {
+      // keyEquivalent is null for SQL Compatible Null Behavior
```

Comment:
I don't understand this comment (same in other classes)

Reply:
this is null only in the case of SQL Compatibility, otherwise null will be replaced with empty string in above line.

Reply:
Whenever I write that I don't understanding something please edit code or add comments, not just explain in the PR. Future code readers won't see those.

Reply:
done.